frontend2/src/views/MyTeam.tsx

```diff
@@ -143,6 +151,36 @@ const MyTeam: React.FC = () => {
           {membersList}
         </SectionCard>
       </div>
+      <Modal
+        isOpen={isLeaveModalOpen}
+        closeModal={() => {
+          setIsLeaveModalOpen(false);
+        }}
+        title="Leave team"
+      >
+        <div className="mt-4 flex flex-col gap-2">
+          <p>
+            Are you sure you want to leave{" "}
+            <span className="font-semibold">{team.name}</span>?
+          </p>
+          <div className="flex flex-row gap-4">
+            <Button
+              variant="danger-outline"
+              onClick={() => {
+                void leaveMyTeam();
```

Comment:
Should we await the leave team function here so the modal only closes once that request has resolved? With a spinner in the button?